araproc/analysis/impulsivity_and_matrices.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.signal import hilbert
+from scipy.stats import linregress
+from scipy.optimize import curve_fit
+from scipy.special import erf
+
+# Define the erf-linear model for curve fitting
+def erf_linear(x, A, B):
+    return (A * erf(x / B) + x) / (A * erf(1 / B) + 1)
+
+# Generate a Gaussian noise waveform (600 ns)
+np.random.seed(42)
+sampling_interval = 0.5  # ns
+time_array = np.arange(0, 600, sampling_interval)
+voltage_array = np.random.normal(0, 1, len(time_array))
+
+# Function to calculate impulsivity-related variables
+def calculate_impulsivity_measures(voltage_array,time_array):
+    result = {}
+
+    # Hilbert transform to get the envelope of the waveform
+    hilbert_envelope = np.abs(hilbert(voltage_array))
+
+    # Find the index of the maximum value in the Hilbert envelope
+    hill_max_idx = np.argmax(hilbert_envelope)
+    hill_max = hilbert_envelope[hill_max_idx]
+
+    # Sorting based on closeness to the maximum index
+    closeness = np.abs(np.arange(len(voltage_array)) - hill_max_idx)
+    clo_sort_idx = np.argsort(closeness)
+
+    # Sort the Hilbert envelope by closeness to the maximum
+    sorted_waveform = hilbert_envelope[clo_sort_idx]
+
+    # Cumulative distribution function (CDF) calculation
+    cdf = np.cumsum(sorted_waveform)
+    cdf /= np.max(cdf)
+
+    # Linear regression to get slope, intercept, and other statistics
+    slope, intercept, r_value, p_value, std_err = linregress(np.arange(len(cdf)), cdf)
+    cdf_fit = slope * np.arange(len(cdf)) + intercept
+    t_frac = np.linspace(0,1,len(cdf))
+
+    # Calculate Kolmogorov-Smirnov statistic 
+    ks = np.max(np.abs(cdf_fit - cdf))
+    # Perform erf-linear fit on the CDF
+    popt, _ = curve_fit(erf_linear, t_frac, cdf, p0=[intercept / slope, 0.01], bounds=(0, [3 * intercept / slope, 0.5]))
```

Comment:
I would place a finite lower bound on parameter B since the function diverges numerically (maybe formally too?) if B is 0. I bounded it to the range (1e-6, 0.5)

Reply:
If I understand you correctly, you are asking for lowering the initial guess of B, doing something like :  popt, _ = curve_fit(erf_linear, t_frac, cdf, p0=[intercept / slope, 1e-6], bounds=(0, [3 * intercept / slope, 0.5])), right?

Reply:
No sorry I think the initial guess was fine, I just mean the lower bound shouldn't be 0. So something like 
`popt, _ = curve_fit(erf_linear, t_frac, cdf, p0=[intercept / slope, 1e-2], bounds=([0,1e-6], [3 * intercept / slope, 0.5]))`